qiita_ware/commands.py

```diff
@@ -228,10 +228,12 @@ def submit_EBI(artifact_id, action, send, test=False, test_size=False):
         cols_to_drop = cols_to_drop - {'taxon_id', 'scientific_name',
                                        'description'}
         all_samples = ebi_submission.sample_template.ebi_sample_accessions
-        samples = {k: all_samples[k] for k in ebi_submission.samples}
-        ebi_submission.write_xml_file(
-            ebi_submission.generate_sample_xml(samples, cols_to_drop),
-            ebi_submission.sample_xml_fp)
+        samples = {k: all_samples[k] for k in ebi_submission.samples
+                   if all_samples[k] is None}
```

Comment:
There's one thing that threw me off when re-reading this PR. The argument `samples` for `generate_sample_xml` is described as a list in the docstring. Should this instead be:

```suggestion
        samples = [k for k in ebi_submission.samples if all_samples[k] is None]
```

Or should the docstring be corrected?

Reply:
Good catch, I'll change. Note that it works cause:
```python
>>> for x in {'a': 1, 'b': 2}:
...    print(x)
... 
a
b
```